web/pdf_scripting_manager.js

```diff
@@ -306,6 +311,10 @@ class PDFScriptingManager {
             pdfViewer.decreaseScale();
           }
           break;
+        case "WillPrintFinished":
+          this.#willPrintResolve?.();
+          this.#willPrintResolve = null;
```

Comment:
With the changes mentioned above, this would become the following.
```suggestion
          this.#willPrintCapability?.resolve();
          this.#willPrintCapability = null;
```

Reply:
Additionally, in the unlikely event that the capability is still pending when destruction happens, please [modify this code as follows](https://github.com/mozilla/pdf.js/blob/7ae5a0fef79703ee5fff9f7dcff21c1922f7be68/web/pdf_scripting_manager.js#L433-L435) to avoid that.
```javascript
    } catch {}

    this.#willPrintCapability?.reject(new Error("Scripting destroyed."));
    this.#willPrintCapability = null;

    for (const [name, listener] of this._internalEvents) {
```